### DISCUSSION TITLE
Server custom structures not readable by every client

### DISCUSSION THREAD

**Discussion by BenjaminKraus4711** [FreeOpcUa/python-opcua#1527] [Q&A]
Since I need to implemnt an opc ua server with several custom structures, I tried out the example (https://github.com/FreeOpcUa/python-opcua/blob/master/examples/server-create-custom-structures.py) and modified it to my needs. 

I tested my server with UAExpert and everything seemed to be fine, finally I tested with another client (Codesys OPC-UA client) and noticed that the structures can't be read - The even do not appear when browsing the address space. I know that the Codesys client is able to read structures (I tested it with the Codesys server). 

I spend some hours investigating potential issues, the only thing I found is that the structures I create have the attribute (IsAbstract=True). The structures in the working Codesys Server set this attribute to False. So I thought this might be the reason but I can't find a way to change the IsAbstract attribute in my server.

Has someone
- An idea how to change the value of this attribute
- Any other ideas why my servers structures can't be read by every client

Thanks in advance

**Comment by schroeder-** [comment]
This library is not supported anymore, switch to [opcua-asyncio](https://github.com/FreeOpcUa/opcua-asyncio) which also has a [sync-wrapper](https://github.com/FreeOpcUa/opcua-asyncio/blob/master/asyncua/sync.py), with very few changes in API. If your problem still exists open a ticktet there.

**Reply by BenjaminKraus4711** [reply]
Thanks for the reply. I know that this library is deprecated, my project is a little bit older and I just wanted to add one custom structure to the existing server without changing the library which would cause a much higher amount of work.

Do you see any chance to add the structure in a proper way or is it unavoidable to switch over to asyncua?
